omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -6,56 +6,38 @@ export default class LatestObsForConceptListController {
+                'groupMembers:(concept:(display,names:(name,locale,localePreferred,voided,conceptNameType))))',
             concept: this.config.concepts.split(',').map(c => c.trim()).join(','),
-            nLatestObs: n_LatestObs
+            nLatestObs: this.config.nLatestObs || 1
         }).then((resp) => {
             // Process the results from the list of concepts as not all of them may have data
-            for (let i = 0; i < resp.results.length; i++) {
-                let obs = resp.results[i];
+            this.obs = resp.results.filter(
                 // Don't add obs older than maxAge
-                if (angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays) {
-                    // Add last obs for concept to list
-                	
-                	if (angular.isDefined(obs.groupMembers) && obs.groupMembers != null) {
-                        // If obs is obs group
-                        let members = [];
-                        angular.forEach(obs.groupMembers, (member) => {
-                        	let prefix;
-                            let value;                            
-                        	
-                        	// Formatting the obs with concept prefix
-                        	if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "FSN") {
-                        		prefix = "(" + member.concept.display + ") ";
-                        	} else if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "shortName") {
-                        		prefix = "(" + member.concept.name.display + ") ";
-                        	} else {
-                                // for default  or obsGroupLabels = none option
-                                prefix = "";
-                        	}
-                            value = this.getObsValue(member);
-                            members.push({"prefix": prefix, "value": value});
-                        });
-                        obs.groupMembers = members;
-
-                    } else {
-                        obs.value = this.getObsValue(obs);
-                    }
-                    this.obs.push(obs);
-                    
+                obs => angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays
+            ).map(inputObs => {
+                const displayObs = {};
+                displayObs.conceptName = this.getConceptName(inputObs.concept, this.config.conceptNameType);
+                if (inputObs.groupMembers) { // If obs is obs group
+                    displayObs.groupMembers = inputObs.groupMembers.map(member => {
+                        const prefix = ["FSN", "shortName", "preferred"].includes(this.config.obsGroupLabels) ?
+                            "(" + this.getConceptName(member.concept, this.config.obsGroupLabels) + ") " : "";
```

Comment:
Group member names can now be  "preferred" (in addition to "FSN" or "shortName").